src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private double firstLeg;
+    private double secondLeg;
+
+    public RightTriangle(String color,double firstLeg, double secondLeg) {
+        super(color);
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+    }
+
+    public double calculateArea() {
+        return (firstLeg * secondLeg) * 0.5;
+    }
```

Comment:
The `calculateArea` method should be part of an interface implementation. According to the checklist, don't put all behavior into a single interface if the methods are conceptually different from each other. Consider creating separate interfaces for `calculateArea` and `drawFigure` methods and implement them in this class.